src/hooks/wallets/mpc/useMPC.ts

```diff
@@ -1,88 +1,82 @@
-import { useEffect } from 'react'
+import { IS_PRODUCTION } from '@/config/constants'
+import { connectWallet, getConnectedWallet } from '@/hooks/wallets/useOnboard'
 import ExternalStore from '@/services/ExternalStore'
-import { COREKIT_STATUS, Web3AuthMPCCoreKit, WEB3AUTH_NETWORK } from '@web3auth/mpc-core-kit'
-import { CHAIN_NAMESPACES } from '@web3auth/base'
-
-import { useCurrentChain } from '@/hooks/useChains'
-import { getRpcServiceUrl } from '../web3'
-import useOnboard, { connectWallet, getConnectedWallet } from '@/hooks/wallets/useOnboard'
-import { useInitSocialWallet } from './useSocialWallet'
+import { SOCIAL_WALLET_OPTIONS } from '@/services/mpc/config'
 import { ONBOARD_MPC_MODULE_LABEL } from '@/services/mpc/SocialLoginModule'
-import { isSocialWalletOptions, SOCIAL_WALLET_OPTIONS } from '@/services/mpc/config'
-import { IS_PRODUCTION } from '@/config/constants'
+import { type ChainInfo } from '@safe-global/safe-gateway-typescript-sdk'
+import { type OnboardAPI } from '@web3-onboard/core'
+import type { Web3AuthMPCCoreKit } from '@web3auth/mpc-core-kit'
+import { COREKIT_STATUS } from '@web3auth/mpc-core-kit'
+import { getRpcServiceUrl } from '../web3'
 
 const { getStore, setStore, useStore } = new ExternalStore<Web3AuthMPCCoreKit>()
 
-export const useInitMPC = () => {
-  const chain = useCurrentChain()
-  const onboard = useOnboard()
-  useInitSocialWallet()
+export const initMPC = async (chain: ChainInfo, onboard: OnboardAPI) => {
+  const chainConfig = {
+    chainId: `0x${Number(chain.chainId).toString(16)}`,
+    chainNamespace: 'eip155',
```

Comment:
Please defne this as a const.

Reply:
We could probably revert this to the constant we were using.